ReSharper.FSharp/src/FSharp.TypeProvidersProtocol/src/Cache/ProvidedAssembliesCache.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq;
+using JetBrains.ReSharper.Plugins.FSharp.TypeProvidersProtocol.Models;
+using JetBrains.ReSharper.Plugins.FSharp.TypeProvidersProtocol.Utils;
+using JetBrains.Rider.FSharp.TypeProvidersProtocol.Server;
+using static FSharp.Compiler.ExtensionTyping;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.TypeProvidersProtocol.Cache
+{
+  internal class ProvidedAssembliesCache : ProvidedEntitiesCacheBase<ProvidedAssembly>
+  {
+    private RdProvidedAssemblyProcessModel ProvidedAssembliesProcessModel =>
+      TypeProvidersContext.Connection.ProtocolModel.RdProvidedAssemblyProcessModel;
+
+    public ProvidedAssembliesCache(TypeProvidersContext typeProvidersContext) : base(typeProvidersContext)
+    {
+    }
+
+    protected override ProvidedAssembly Create(int key, int typeProviderId, ProvidedTypeContextHolder context)
+      => ProxyProvidedAssembly.Create(
+        TypeProvidersContext.Connection.ExecuteWithCatch(() =>
+          ProvidedAssembliesProcessModel.GetProvidedAssembly.Sync(key)),
+        TypeProvidersContext.Connection);
+
+    protected override ProvidedAssembly[] CreateBatch(int[] keys, int typeProviderId, ProvidedTypeContextHolder context)
+      => throw new System.NotImplementedException();
```

Comment:
_(Minor)_ `NotSupportedException` is better for this case, I believe. `NotImplementedException` is more like a TODO (which is appropriate if we plan to implement this method one day, which I don't think is the case here).